### PR TITLE
PR for #30 - Include function arguments in function name

### PR DESCRIPTION
**Pull request by chess-equality**
#30

### CODE REVIEW

**Comment by BFergerson** [comment]
@chess-equality, can you remove all the anonymous function code first before I review? Thanks.

File: src/main/groovy/com/codebrig/arthur/observe/structure/filter/FunctionFilter.groovy [hunk 1]
```diff
@@ -15,9 +15,22 @@ class FunctionFilter extends StructureFilter<FunctionFilter, Void> {
     private final MultiFilter filter
 
     FunctionFilter() {
-        filter = MultiFilter.matchAll(
-                new RoleFilter("DECLARATION"), new RoleFilter("FUNCTION"),
-                new RoleFilter().reject("ARGUMENT", "RETURN", "INCOMPLETE", "BODY")
+        filter = MultiFilter.matchAny(
```
Comment (BFergerson) [inline src/main/groovy/com/codebrig/arthur/observe/structure/filter/FunctionFilter.groovy]: You're going to have to revert the change to this filter and remove the tests that fail. It looks like you've added support for dynamically created functions for JavaScript/Ruby and that's not the point of this filter. In the future, there may be a `DynamicFunctionFilter` or something but this filter is not it.

If you noticed, I didn't include any dynamically created methods for Java. I could have but right now I'm just worried about static functions.

File: src/main/groovy/com/codebrig/arthur/observe/structure/filter/NameFilter.groovy [hunk 2]
```diff
@@ -24,9 +24,27 @@ class NameFilter extends StructureFilter<NameFilter, String> {
...
+                    case "FunctionExpression":
+                    case "ArrowFunctionExpression":
+                        childNameFilter = new InternalRoleFilter("id").getFilteredNodes(node.children)
+                        break
+                    case "ObjectMethod":
+                        childNameFilter = MultiFilter.matchAll(
+                                new TypeFilter("Identifier"), new InternalRoleFilter("key")
+                        ).getFilteredNodes(node.children)
+                        break
+                    case "NewExpression":
```
Comment (BFergerson) [inline src/main/groovy/com/codebrig/arthur/observe/structure/filter/NameFilter.groovy]: I think the same deal here. This filter isn't meant to handle dynamically created functions. I created an issue to address it at some point in the future (#65).

Comment (chess-equality) [reply src/main/groovy/com/codebrig/arthur/observe/structure/filter/NameFilter.groovy]: @BFergerson Acknowledged thanks, will commit changes later today.

**Comment by chess-equality** [comment]
@BFergerson Please see my latest commit. Thanks.

**Comment by BFergerson** [comment]
@chess-equality, this looks good. Thanks for opening those Babelfish issues. I've subscribed to them all.

**Pull request merged**